script.js

```diff
@@ -2,37 +2,42 @@
 let quizQuestions = [];
 
 // Function to create an object from the quiz question
-function createQuizQuestion() {
-    const question = document.getElementById("question").value;
-
-    // Select all text inputs within .option-container dives
-    const optionInputs = document.querySelectorAll(
-        ".option-container input[type='text']"
-    );
-
-    // Convert NodeList to an array and map to their values
-    const options = Array.from(optionInputs).map((input) => input.value);
-
-    const correctIndex =
-        document.querySelector('input[name="correct"]:checked').value - 1; // The radios value starts from number 1
-    const explanation = document.getElementById("explanation").value;
-
-    // According to the format of the object of the quiz question
-    const quizQuestion = {
-        id: quizQuestions.length + 1, // To make a new id for each quiz question
-        question: question,
-        options: options.map((option, index) => ({
-            text: option,
-            isCorrect: index === correctIndex, // Mark as true if this option corresponds to the selected radio button
-        })),
-        explanation: explanation,
-    };
-
-    quizQuestions.push(quizQuestion);
-
-    console.log(quizQuestions);
-    alert("Quiz question submitted successfully!");
-}
+document
+    .getElementById("quiz-form")
+    .addEventListener("submit", function (event) {
```

Comment:
I would abstract this `function (event) {...}` to a separate onSubmit function and then call it here.